firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java

```diff
@@ -256,10 +263,11 @@ public HttpURLConnection createRealtimeConnection() throws IOException {
 
   /** Retries HTTP stream connection asyncly in random time intervals. */
   @SuppressLint("VisibleForTests")
-  public synchronized void retryHTTPConnection() {
+  public synchronized void tryHttpConnection(long retrySeconds) {
     if (canMakeHttpStreamConnection() && httpRetriesRemaining > 0) {
-      if (httpRetriesRemaining < ORIGINAL_RETRIES) {
-        httpRetrySeconds *= getRetryMultiplier();
+      if (httpRetriesRemaining < ORIGINAL_RETRIES && retrySeconds > 0) {
+        retrySeconds *= getRetryMultiplier();
```

Comment:
It's a bit confusing to have both retry seconds passed into the method and stored as an instance variable. Maybe some more specific naming could clear up why we have both? What about a boolean param or two methods to separate the first try and retry logic?

Reply:
Updated to have two methods, one that starts the initial stream connection and one fore retrying